pkg/googlecloud/googlecloud.go

```diff
@@ -99,4 +103,4 @@ func (m *googleCloudClient) GetCurrentNodeCount(clusterID string) (int32, error)
 }
 
 // Make sure the real implementation complies with its interface.
-var _ interfaces.GoogleCloudClient = (*googleCloudClient)(nil)
+var _ GoogleCloudClient = (*googleCloudClient)(nil)
```

Comment:
Do we need this to ensure the client implements the interface? `ClientBuilder` initializes the struct and needs to explicitly return a `GoogleCloudClient`.

```suggestion
```

Reply:
This ensures that the instance implements all methods required by the interface.
We could remove this, but having something like that helps you to understand early what you did wrong.
I see this as a good practice.

Reply:
> We could remove this, but having something like that helps you to understand early what you did wrong.

What I meant is that maybe Golang will ensure this for you by static type checking :thinking: